src/components/installments/InstallmentPaymentStateInfo.tsx

```diff
@@ -0,0 +1,143 @@
+import classNames from 'classnames';
+import React, { useCallback } from 'react';
+import {
+  Button,
+  Dialog,
+  IconErrorFill,
+  IconAlertCircleFill,
+  IconCheckCircleFill,
+  IconInfoCircle,
+  Table,
+} from 'hds-react';
+import { useTranslation } from 'react-i18next';
+
+import formattedCurrency from '../../utils/formatCurrency';
+import { ApartmentInstallmentPayment, ApartmentInstallmentPaymentState } from '../../types';
+import formatDateTime from '../../utils/formatDateTime';
+
+import styles from './InstallmentPaymentStateInfo.module.scss';
+
+const T_PATH = 'components.installments.InstallmentPaymentStateInfo';
+
+interface IProps {
+  state: ApartmentInstallmentPaymentState;
+  payments: ApartmentInstallmentPayment[];
+  sentToSap: boolean;
+}
+
+type StatusLevel = 'unimportant' | 'success' | 'important' | 'critical';
+
+const statusLevel = function statusLevel(state: ApartmentInstallmentPaymentState): StatusLevel {
+  if (state.status === 'PAID') {
+    if (state.overdue) {
+      return 'important';
+    }
+    return 'success';
+  } else if (state.overdue || state.status === 'OVERPAID') {
+    return 'critical';
+  } else if (state.status === 'UNPAID') {
+    // UNPAID and not overdue is unimportant
+    return 'unimportant';
+  }
+  // Rank UNDERPAID as important
+  return 'important';
+};
+
+const statusIcon = function statusIcon(level: StatusLevel): JSX.Element | null {
+  switch (level) {
+    case 'critical':
+      return <IconErrorFill />;
+    case 'important':
+      return <IconAlertCircleFill />;
+    case 'success':
+      return <IconCheckCircleFill />;
+    case 'unimportant':
+      return null;
+  }
+};
+
+const InstallmentPaymentStateInfo = ({ state, payments, sentToSap }: IProps) => {
+  const { t: translate } = useTranslation();
+  const t = (label: string) => translate(`${T_PATH}.${label}`);
+
+  const showPayments = payments?.length || null;
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const open = () => {
+    if (showPayments) {
+      setIsOpen(true);
+    }
+  };
+  const close = useCallback(() => {
+    setIsOpen(false);
+  }, []);
+
+  let messages = [];
+
+  if (!sentToSap) {
+    messages.push(t('unsent'));
+  } else {
+    messages.push(translate(`ENUMS.InstallmentPaymentStatus.${state.status}`));
+  }
+  if (state.overdue) {
+    messages.push(t('overdue'));
+  }
+
+  const level = statusLevel(state);
+
+  const mappedPayments = showPayments
+    ? payments.map((p, i) => ({
+        index: i,
+        amount: formattedCurrency(p.amount / 100),
+        payment_date: formatDateTime(p.payment_date, true),
+      }))
+    : [];
+
+  return (
+    <>
+      <div
+        className={classNames(styles.container, styles[`status-${level}`], showPayments && styles.clickable)}
+        onClick={open}
+      >
```

Comment:
It's not a good idea to use plain `<div>` as a clickable element without having defined it as a button and being able to focus it with a keyboard. I would render the wrapping element based on the `showPayments` value.

If there's something to open, then render an anchor element or a button. And if not, then just a plain `<div>`